src/coordinator/storage/converter.go

```diff
@@ -177,3 +183,90 @@ func SeriesToPromSamples(series *ts.Series) []*prompb.Sample {
 
 	return samples
 }
+
+const (
+	initRawFetchAllocSize = 32
+	workerPoolSize        = 10
+)
+
+var (
+	workerPool sync.WorkerPool
+)
+
+func init() {
+	// Set up worker pool
+	workerPool = sync.NewWorkerPool(workerPoolSize)
+	workerPool.Init()
+}
+
+type decompressRequest struct {
+	iter      encoding.SeriesIterator
+	namespace ident.ID
+	result    *ts.Series
+}
+
+func (w *decompressRequest) Process(ctx context.Context) error {
```

Comment:
nit: add a comment here

Reply:
Surprised the linter let me get away with that